core/circuitbreaker/rule_manager.go

```diff
@@ -315,3 +312,25 @@ func RemoveCircuitBreakerGenerator(s Strategy) error {
 	delete(cbGenFuncMap, s)
 	return nil
 }
+
+func IsValid(r *Rule) error {
+	if len(r.Resource) == 0 {
+		return errors.New("empty resource name")
+	}
+	if int(r.Strategy) < int(SlowRequestRatio) || int(r.Strategy) > int(ErrorCount) {
+		return errors.New("invalid Strategy")
+	}
+	if r.StatIntervalMs <= 0 {
+		return errors.New("invalid StatIntervalMs")
+	}
+	if r.Threshold < 0 {
+		return errors.New("invalid Threshold")
+	}
+	if r.Strategy == SlowRequestRatio && (r.Threshold < 0.0 || r.Threshold > 1.0) {
```

Comment:
`r.Threshold < 0.0` is not needed as it has been checked before.

Reply:
updated